torch/autograd/graph.py

```diff
@@ -370,11 +384,18 @@ def disable_saved_tensors_hooks(error_message):
 
 def register_multi_grad_hook(
     tensors: Sequence[torch.Tensor],
-    fn: Callable[[Sequence[Optional[torch.Tensor]]], None],
+    fn: Union[
```

Comment:
@soulitzer The typing is kind of cooked because `mode="all"` and `mode="any"` accept different `fn` signatures.

Reply:
Ah welp, I guess we will have to live with that lol 

This makes me wonder if an extra sentence in the docs would be slightly clear:
after ``a tensor in :attr:`tensors` has been computed....``
maybe add something like `The hook will be called with the single gradient that has been computed`

Reply:
I added a sentence 😄